### PR TITLE
[Docs] Pin to specific version of Python in the example

### PR DESCRIPTION
**Pull request by dimakuv**
## Description of the changes 

Python default (`latest`) Docker image recently switched to Debian 12 which uses PEP668 (marking Python base environments as "externally managed"). This change is not yet supported by GSC, so let's pin the Python example to a supported version.

See also https://github.com/gramineproject/gsc/issues/153 and https://peps.python.org/pep-0668/

## How to test this PR? 

Manually test that Python example works.

- - -
This change is [Reviewable](https://reviewable.io/reviews/gramineproject/gsc/158)

### CODE REVIEW

**Review by kailun-qin — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gsc/158)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel)

**Review by mkow — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gsc/158)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @dimakuv)

___
*[`-- commits` line 7 at r1](https://reviewable.io/reviews/gramineproject/gsc/158#-N_3hJKWEvXnDY4C9J5p):*
How hard would it be to fix the issue instead of pinning to an older version? Using older software is a bit fishy practice :) And also easy to forget to update the version later, so if the real fix isn't too hard then I'd prefer it instead.

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gsc/158)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @woju)

___
*[`-- commits` line 7 at r1](https://reviewable.io/reviews/gramineproject/gsc/158#-N_3hJKWEvXnDY4C9J5p):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

How hard would it be to fix the issue instead of pinning to an older version? Using older software is a bit fishy practice :) And also easy to forget to update the version later, so if the real fix isn't too hard then I'd prefer it instead.
</blockquote></details>

The proper fix is here: https://github.com/gramineproject/gsc/pull/155

In other words, to support the latest Python version we need Debian 12 support, which in turn needs a resolution of the PEP668 problem.

However, I don't think #155 is trivial, and we'll need to wait for @woju to give his opinion... So I created this PR to go with the workaround, until the PEP668 issue is fixed.

Also, isn't it generally a good practice to pin versions of software?

**Review by mkow — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gsc/158)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required)

___
*[`-- commits` line 7 at r1](https://reviewable.io/reviews/gramineproject/gsc/158#-N_3hJKWEvXnDY4C9J5p):*
Ok then.

> Also, isn't it generally a good practice to pin versions of software?

Hmm, I think it depends on the specific case. It gives you more confidence that something will keep working and won't break spontaneously, but you may also miss important updates if that specific component isn't actively maintained.

**Review by kailun-qin — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gsc/158)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by mkow — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gsc/158)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gsc/158)* status: :shipit: complete! all files reviewed, all discussions resolved

*[a discussion](https://reviewable.io/reviews/gramineproject/gsc/158#-N_44ysmEndKs9S5QcsQ:-N_44ysmEndKs9S5QcsR:b-4xc48i) (no related file):*
Did a final rebase and manually verified that Python example works.

___
